src/service/repository/challenge.rs

```diff
@@ -140,4 +140,16 @@ impl Challenge {
                 FailedToGetChallenge(GetChallengeError(e)).into()
             })
     }
+
+    pub fn delete(connection: &mut PgConnection, challenge_id: &Uuid) -> Result<usize> {
```

Comment:
No need for a return type if we're not returning anything based on my recommendation.

```suggestion
    pub fn delete(connection: &mut PgConnection, challenge_id: &Uuid) -> Result<()> {
```